pull.sh

```diff
@@ -193,6 +193,10 @@ update () {
 
   local curbranch="$(git branch --show-current)"
 
+  # do an initial push just to make sure it exists, otherwise
+  # the retargetting will fail.
+  git push origin "$curbranch"
```

Comment:
Actually the bit that pushes to the current branch has extra logic to check if we're currently sitting in a default branch in order to avoid pushing (just wanted to be extra safe in that case, ref: https://github.com/npm/pull/pull/5/files#diff-df027a58eb34e6f4541846cb4f7be10824a342ab71b4f006a18de21a14f90a76R215)

So for the sake of sticking to that idea I think it's better to also wrap this `git push` in a similar way:

```suggestion
  if [ "$curbranch" != "" ] || \
     [ "$curbranch" != "$defaultbranch" ] || \
     [ "$curbranch" != "master" ] || \
     [ "$curbranch" != "main" ] || \
     [ "$curbranch" != "latest" ]; then
    git push origin "$curbranch"
  fi
```

Reply:
either that OR just being fine with it pushing to master/latest/main and removing the conditional logic from the later `git push`

Reply:
Ahh, ok.  Good call.

Maybe we need a `should_push "$branch"` function, so we could just do:

```bash
if should_push "$curbranch"; then
  git push origin $curbranch
fi
```